src/apphosting/backend.ts

```diff
@@ -340,6 +388,7 @@ async function provisionDefaultComputeServiceAccount(projectId: string): Promise
       "roles/firebaseapphosting.computeRunner",
       "roles/firebase.sdkAdminServiceAgent",
       "roles/developerconnect.readTokenAccessor",
+      "roles/storage.objectViewer",
```

Comment:
I think this is always added even in non source deploy cases whenever there is a 404? Why is that?

Reply:
It's added in case the customer wants to use deploy the source in the future - if we can add the storage.objectViewer role now, then they won't run into the IAM propagation delay problem in the future.